### ISSUE TITLE
Storing additional user info in session data

### ISSUE THREAD

**Issue by dvalley56** [supertokens/supertokens-node#403]
I want to store user's first name and last name with userId and email in session (`for use only in backend`).

to implement this and found few solutions.

I looked in the docs , querying user data and store in session during `createNewSession`, it works for middlewares very well but getting that session data in the pre/post hooks doesn't seems correct.

`This is how I got that data in hooks`
![image](https://user-images.githubusercontent.com/67868355/195970079-85307907-451a-4eb5-aa31-3bde4e5f056a.png)

Is there another approach to access this in hooks or there is something else where I should store use data ?

I want this data only in backend for cases like sending email to the user and  using their first name to greet them.

**Comment by rishabhpoddar** [comment]
Hey @dvalley56 

Did you add the name and email to the accessTokenPayload or to the sessionData?

If you added it to the accessTokenPayload, then you should do `b. accessTokenPayload`

**Comment by rishabhpoddar** [comment]
Closing this issue due to non responsiveness. Please feel free to reopen

**Comment by dvalley56** [comment]
I am sorry I didn't reply.
I added it in the sessionData by overriding `createNewSession`.

```
createNewSession: async (input) => {

  const user: any = await dbConnection(
    'SELECT email, first_name as firstName, last_name as lastName FROM mytable WHERE user_id = ?',
    [input.userId]
  );


  input.sessionData = {
    ...input.sessionData,
    user: user[0],
  };

  return originalImplementation.createNewSession(input);
}
```

This is what I do inside the block.

**Comment by rishabhpoddar** [comment]
Do you see the data added in the database table (session_info) after `originalImplementation.createNewSession(input);` has executed?

**Comment by rishabhpoddar** [comment]
@dvalley56 would appreciate your input here. Thanks

**Comment by dvalley56** [comment]
> @dvalley56 would appreciate your input here. Thanks

Yes, the data is being added to the table `session_info` under the column `session _data`. Also I am getting the data as expected by the code I shared ablove.

`I am so sorry mate for ghosting the issue, currently I have my semester's exam going on so I didn't check here.`

**Comment by rishabhpoddar** [comment]
The approach you used is correct. Im closing this issue for now since it's not really a bug in SuperTokens
